File: benchmarks/execution/JS/Sieve.ts
import {BInteger} from './btypes/BInteger.js';
import {BBoolean} from './btypes/BBoolean.js';
import {BSet} from './btypes/BSet.js';
import {BUtils} from "./btypes/BUtils.js";
import {SelectError} from "./btypes/BUtils.js";



export default class Sieve {




    private numbers: BSet<BInteger>;
    private cur: BInteger;
    private limit: BInteger;

    constructor() {
        this.numbers = BSet.interval(new BInteger(2), new BInteger(1000000));
        this.cur = new BInteger(2);
        this.limit = new BInteger(1000000);
    }


     ComputeNumberOfPrimes(): BInteger {
        let res: BInteger = null;
        while((new BBoolean(this.cur.greater(new BInteger(1)).booleanValue() && this.cur.multiply(this.cur).lessEqual(this.limit).booleanValue())).booleanValue()) {
            if((this.numbers.elementOf(this.cur)).booleanValue()) {
                let n: BInteger = null;
                let set: BSet<BInteger> = null;
                n = this.cur;
                set = new BSet<BInteger>();
                while((n.lessEqual(this.limit.divide(this.cur))).booleanValue()) {
                    set = set.union(new BSet<BInteger>(this.cur.multiply(n)));
                    n = n.plus(new BInteger(1));
                }
                this.numbers = this.numbers.difference(set);
            } 
            this.cur = this.cur.plus(new BInteger(1));
        }
        res = this.numbers.card();
        return res;
    }

    _get_numbers(): BSet<BInteger> {
        return this.numbers;
    }

    _get_cur(): BInteger {
        return this.cur;
    }

    _get_limit(): BInteger {
        return this.limit;
    }


}

let sieve: Sieve = new Sieve();
console.log(sieve.ComputeNumberOfPrimes().toString());

File: benchmarks/execution/Java_Clojure_Set_Primitives/Sieve.java
import de.hhu.stups.btypes.BSet;
import de.hhu.stups.btypes.BInteger;
import de.hhu.stups.btypes.BBoolean;
import java.util.Objects;
import java.util.Arrays;
import de.hhu.stups.btypes.PreconditionOrAssertionViolation;
import de.hhu.stups.btypes.StateNotReachableError;
import de.hhu.stups.btypes.BUtils;


public class Sieve {







    private BSet<BInteger> numbers;
    private BInteger cur;
    private BInteger limit;

    public Sieve() {
        numbers = BSet.interval(new BInteger(2), new BInteger(10000));
        cur = new BInteger(2);
        limit = new BInteger(10000);
    }

    public BInteger ComputeNumberOfPrimes() {
        BInteger res = null;
        while((new BBoolean(cur.greater(new BInteger(1)).booleanValue() && cur.multiply(cur).lessEqual(limit).booleanValue())).booleanValue()) {
            if((numbers.elementOf(cur)).booleanValue()) {
                BInteger n = null;
                BSet<BInteger> set = null;
                n = cur;
                set = new BSet<BInteger>();
                while((n.lessEqual(limit.divide(cur))).booleanValue()) {
                    set = set.union(new BSet<BInteger>(cur.multiply(n)));
                    n = n.plus(new BInteger(1));
                }
                numbers = numbers.difference(set);
            } 

            cur = cur.plus(new BInteger(1));
        }
        res = numbers.card();
        return res;
    }

    public BSet<BInteger> _get_numbers() {
        return numbers;
    }

    public BInteger _get_cur() {
        return cur;
    }

    public BInteger _get_limit() {
        return limit;
    }


    public static void main(String[] args) {
        Sieve sieve = new Sieve();
        System.out.println(sieve.ComputeNumberOfPrimes());
    }

}
